### PR TITLE
Improve GitHub Actions and add one for indexer

### PR DESCRIPTION
**Pull request by JakeSCahill**

### CODE REVIEW

File: .github/workflows/build-site.yml [hunk 1]
```diff
@@ -16,9 +16,12 @@ jobs:
...
-    - run: npm i
-    - run: npm run-script build
-    - run: npm run-script check-links
+    - run: |
+        git config --global user.name 'JakeSCahill'
+        git config --global user.email '45230295+JakeSCahill@users.noreply.github.com'
+        npm i
+        npm run-script build
+        touch docs/.nojekyll
+        git add -f docs
```
Comment (pivovarit) [inline .github/workflows/build-site.yml:25]: What about the Github Pages deploy extension?

https://github.com/pivovarit/articles/blob/master/.github/workflows/reveal-pdf.yml#L23-L29

Comment (JakeSCahill) [reply .github/workflows/build-site.yml:25]: It definitely looks interesting. I'm just not sure at the moment. It's quite nice to be able to manually decide when to publish. We may want to deploy it on a staging site first for example.

Comment (pivovarit) [reply .github/workflows/build-site.yml:25]: Keep in mind that the above extension is just something that allows you to abstract out manual git operations so you might find it useful anyway. GH Pages deployment is just a commit to a designated branch

Comment (JakeSCahill) [reply .github/workflows/build-site.yml:25]: Oh I see. So could we potentially create a publish job that is called only when content in another repo is merged?

Comment (pivovarit) [reply .github/workflows/build-site.yml:25]: It's a good question. Certainly, you could rely on such a trigger within a single repository, but if you would like to do this cross-repo, I'm afraid you'd need to use the eventing system:

https://github.community/t/triggering-by-other-repository/16163

**Pull request merged**